packages/x-date-pickers/src/DatePicker/DatePicker.tsx

```diff
@@ -12,6 +12,14 @@ type DatePickerComponent = (<TDate>(
   props: DatePickerProps<TDate> & React.RefAttributes<HTMLDivElement>,
 ) => React.JSX.Element) & { propTypes?: any };
 
+/**
+ * Demos:
```

Comment:
At minimum we need a new line here to avoid the unwanted behavior of `docs:api` parsing this as `componentDescription`. 🙈 
![Screenshot 2023-10-10 at 19 43 57](https://github.com/mui/mui-x/assets/4941090/39efddca-8d56-4e42-9904-0f2a29d21b00)

```suggestion
 * Demos:

```

Reply:
Would it help if we use the `@demos` annotation here instead?
I used it in a few places in the data grid:

https://github.com/mui/mui-x/blob/5cd78728893877a73c6d8b1088296ad636431fa8/packages/grid/x-data-grid/src/hooks/features/statePersistence/gridStatePersistenceInterface.ts#L28-L29

This annotation is not listed in https://jsdoc.app/, so I'm not sure how would different IDEs treat it, but here is how it looks like in VS Code:

<img width="428" src="https://github.com/mui/mui-x/assets/13808724/485843b2-f287-4d42-9670-6a45d56925b2">

We could also consider using `@tutorial` annotation: https://jsdoc.app/tags-inline-tutorial.html

Reply:
It would probably become a general company level question, because now the idea is to align the comments with the approach taken by MUI Material. 🤔

Reply:
The core generates this comment based on the doc generated files as far as I'm aware
If we do the opposite I think it would create quite a lot of confusion

Reply:
Yes, for curious people, there are the line adding those comments:

https://github.com/mui/material-ui/blob/master/packages/api-docs-builder/ApiBuilders/ComponentApiBuilder.ts/#L286-L318

Reply:
oh, ok ... so the JSDoc block gets inserted automatically. This makes a lot of sense, so we can probably mimic this behaviour or simply use the existing one from core, right?

Reply:
Good luck, it's a mix of AST manipulation and some naming conventions that are not explicitly defined and probably not respected by X 🫣

Reply:
ok, so @LukasTy was right 🙇🏼 
Adding a line in between `Demos:`/`API:` and the list removes it from the `componentDescription`